src/libraries/ProtocolFeeLibrary.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity ^0.8.24;
+
+import "./math/UnsafeMath.sol";
+
+library ProtocolFeeLibrary {
+    // Max protocol fee is 0.1% (1000 pips)
+    uint16 public constant MAX_PROTOCOL_FEE = 1000;
+
+    // the protocol fee is represented in hundredths of a bip
+    uint256 internal constant PIPS_DENOMINATOR = 1_000_000;
+
+    function getZeroForOneFee(uint24 self) internal pure returns (uint16) {
+        return uint16(self & (4096 - 1));
+    }
+
+    function getOneForZeroFee(uint24 self) internal pure returns (uint16) {
+        return uint16(self >> 12);
+    }
+
+    function validate(uint24 self) internal pure returns (bool) {
+        if (self != 0) {
+            uint16 fee0 = getZeroForOneFee(self);
+            uint16 fee1 = getOneForZeroFee(self);
+            // The fee is represented in pips and it cannot be greater than the MAX_PROTOCOL_FEE.
+            if ((fee0 > MAX_PROTOCOL_FEE) || (fee1 > MAX_PROTOCOL_FEE)) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
(optional)
just revert instead of returning boolean for simplicity and also to align with lp fee validate